mynn/layers/batchnorm.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import mygrad as mg
+
+from mynn.initializers.constant import constant
+
+class batchnorm:
+    ''' A batch normalization layer.
+
+    This class will perform an n-dimensional batch normalization operation on an (N, D, ...)
+    shaped tensor scaled by γ of shape (D, ...) and shifted by β of shape (D, ...).
+    '''
+    def __init__(self, input_shape, momentum=0.1, gamma_initializer=constant, 
+                 beta_initializer=constant, moving_mean_initializer=constant, 
+                 moving_var_initializer=constant, gamma_kwargs={"value": 1.0}, 
+                 beta_kwargs=None):
+        ''' Initialize a batch normalization layer.
+
+        Parameters
+        ----------
+        input_shape : Sequence[int]
```

Comment:
We don't need the full shape; we can do this with just having the number of input channels, which is much cleaner